service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderServiceTest.java

```diff
@@ -1,150 +1,257 @@
-        return baseDirection(PARENTS_AND_RESPONDENTS, respondent, null);
+    private CaseManagementOrder createCaseManagementOrder() {
+        return CaseManagementOrder.builder()
+            .hearingDate(formatLocalDateToMediumStyle(2))
+            .id(fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2"))
+            .recitals(wrapElements(Recital.builder().build()))
+            .schedule(Schedule.builder().build())
+            .status(SELF_REVIEW)
+            .orderDoc(DocumentReference.builder().build())
+            .build();
+    }
+
+    private String formatLocalDateToMediumStyle(int i) {
+        return formatLocalDateToString(time.now().plusDays(i).toLocalDate(), FormatStyle.MEDIUM);
     }
 
-    private Direction expectedDirection(OtherPartiesDirectionAssignee other) {
-        return baseDirection(OTHERS, null, other);
+    @Nested
+    class PrepareCaseDetailsTest {
+        private final String[] keys = {
+            HEARING_DATE_LIST.getKey(),
+            RECITALS.getKey(),
+            SCHEDULE.getKey()};
+
+        private HashMap<String, Object> data; // Tries to use an ImmutableMap unless specified
+
+        @Test
+        void shouldRemoveScheduleAndRecitalsAndHearingDateListFromCaseData() {
+            data = new HashMap<>();
+
+            Arrays.stream(keys).forEach(key -> data.put(key, ""));
+
+            service.removeTransientObjectsFromCaseData(data);
+
+            assertThat(data).doesNotContainKeys(keys);
+        }
     }
 
-    private Direction expectedDirection(DirectionAssignee assignee) {
-        return baseDirection(assignee, null, null);
+    @Test
+    void shouldGetCaseManagementOrderWithDocumentReference() {
+        initMocks();
+
+        LocalDateTime dateTime = LocalDateTime.of(2099, 1, 1, 10, 0, 0);
+        Document document = service.getOrder(buildCaseDataForCMODocmosisGeneration(dateTime));
+
+        assertThat(document).isEqualTo(DOCUMENT);
     }
 
-    private Direction baseDirection(DirectionAssignee assignee,
-                                    ParentsAndRespondentsDirectionAssignee respondent,
-                                    OtherPartiesDirectionAssignee other) {
-        return Direction.builder()
-            .directionType("Direction title")
-            .directionText("Mock direction text")
-            .directionNeeded(YES.getValue())
-            .assignee(assignee)
-            .parentsAndRespondentsAssignee(respondent)
-            .otherPartiesAssignee(other)
-            .readOnly("No")
-            .custom("Yes")
-            .dateToBeCompletedBy(LocalDateTime.of(2099, 1, 1, 10, 0, 0))
-            .build();
+    private void initMocks() {
+        DocmosisOrder order = any(DocmosisOrder.class);
+        given(documentService.getDocumentFromDocmosisOrderTemplate(order, eq(CMO))).willReturn(DOCUMENT);
+        given(templateDataService.getTemplateData(any())).willReturn(DocmosisCaseManagementOrder.builder().build());
```

Comment:
I think I would prefer having this on the top, I know it's not really common to have private on the top, but I would like see easily how mine mocks were setup I think it's important.

Thoughts?